articles/quickstart/spa/vuejs/_includes/_centralized_login.md

```diff
@@ -158,50 +228,230 @@ Using Universal Login means that users are taken away from your application to a
 
 When a user authenticates at the login page and is then redirected back to your application, their authentication information will be contained in a URL hash fragment. The `handleAuthentication` method in the `AuthService` is responsible for processing the hash.
 
-Create a component named `CallbackComponent` and populate it with a loading indicator. The component should also call `handleAuthentication` from the `AuthService`.
+Install `vue-router` to allow callbacks to be routed properly to the `Callback` component:
+
+```bash
+npm install vue-router
+```
+
+Add a new file `router.js` inside the `src` folder with the following content:
+
+```js
+// src/router.js
+
+import Router from 'vue-router';
+import Callback from './components/Callback';
+
+const routes = [
+  {
+    path: '/callback',
```

Comment:
I've also changed the order of steps here. The callback component is now created before the router (where the component is imported) so there's less risk of the reader getting compiler errors because they haven't completed the following steps.